examples/common/text_data.py

```diff
@@ -137,13 +137,65 @@ def __getitem__(self, idx: int) -> Dict[str, Any]:
         return token_sample
 
 
+class ConcatenatedSequenceCollatorWrapper:
+    """Collator wrapper to add sequence_id to batch."""
+
+    def __init__(self,
+                 base_collator: Callable,
+                 eos_token_id: Optional[int] = None,
+                 bos_token_id: Optional[int] = None):
+        self.base_collator = base_collator
+        if (eos_token_id is None) and (bos_token_id is None):
+            raise ValueError(
+                'Must supply a value for either eos_token_id or bos_token_id, but got None for both.'
+            )
+        if (eos_token_id is not None) and (bos_token_id is not None):
+            raise ValueError(
+                'Must supply a value for either eos_token_id or bos_token_id, but both were given.'
+            )
+        if eos_token_id is None:
+            self.split_token_id = bos_token_id
+            self.eos_mode = False
+        else:
+            self.split_token_id = eos_token_id
+            self.eos_mode = True
```

Comment:
Why use `self.split_token_id` and `self.eos_mode`?
Why not just set `self.eos_token_id` and `self.bos_token_id` then in call make the call based on those?